pkg/playback/http_serializer.go

```diff
@@ -0,0 +1,92 @@
+package playback
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+type serializer func(input interface{}) (bytes []byte, err error)
+type deserializer func(input *io.Reader) (value interface{}, err error)
+
+func httpRequestToBytes(input interface{}) (data []byte, err error) {
+	return json.Marshal(input)
+}
+
+func httpRequestfromBytes(input *io.Reader) (val interface{}, err error) {
+	output := httpRequest{}
+
+	inputBytes, err := ioutil.ReadAll(*input)
+	if err != nil {
+		return output, err
+	}
+
+	err = json.Unmarshal(inputBytes, &output)
+	return output, err
+}
+
+func httpResponseToBytes(input interface{}) (data []byte, err error) {
+	return json.Marshal(input)
+}
+
+func httpResponsefromBytes(input *io.Reader) (val interface{}, err error) {
+	output := httpResponse{}
+
+	inputBytes, err := ioutil.ReadAll(*input)
+	if err != nil {
+		return output, err
+	}
+
+	err = json.Unmarshal(inputBytes, &output)
+	return output, err
+}
+
+type httpResponse struct {
+	Headers    http.Header
+	Body       []byte
+	StatusCode int
+}
+
+func newHTTPResponse(resp *http.Response) (wrappedResponse httpResponse, err error) {
+	wrappedResponse = httpResponse{}
+
+	wrappedResponse.Headers = resp.Header
+	wrappedResponse.StatusCode = resp.StatusCode
+
+	var bodyBytes []byte
+	bodyBytes, err = ioutil.ReadAll(resp.Body)
+	defer resp.Body.Close()
+	if err != nil {
+		return wrappedResponse, err
+	}
+	wrappedResponse.Body = bodyBytes
+
+	return wrappedResponse, nil
+}
+
+type httpRequest struct {
+	Method  string
+	Body    []byte
+	Headers http.Header
+	URL     url.URL
+}
+
+func newHTTPRequest(req *http.Request) (wrappedRequest httpRequest, err error) {
```

Comment:
A docstring explaining the "why" of this file would be good, too.

Reply:
Done! 85caf865a9d8b828a9635da735339795656f2173